pallets/template/src/lib.rs

```diff
@@ -63,6 +82,36 @@ decl_module! {
 		// Events must be initialized if they are used by the pallet.
 		fn deposit_event() = default;
 
+		fn on_finalize(current_block_number: T::BlockNumber) {
+            debug::info!("current block number {:#?}", current_block_number);
+
+			let something_count = Self::something_count();
+
+			<SomethingCount<T>>::put(something_count + 1u32.into()); // One::one() or 1u32.into()
+
+			// PROBLEM 1: WHAT DO I HAVE TO DO TO MAKE THIS COMPILE? PLEASE EXPLAIN SOLUTION SO I UNDERSTAND
+			// ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ----------- this method call resolves to `T`
+			// cannot infer type for type parameter `KeyArg` declared on the associated function `insert`
+			// note: cannot satisfy `_: EncodeLike<<T as Config>::SomethingConfigIndex>`
+			// note: required by `hidden_include::StorageMap::insert`
+			// <SomethingConfig<T>>::insert(0u64.into(), current_block_number);
```

Comment:
```Rust
let index: T::SomethingConfigIndex = 0u32.into();
<SomethingConfig<T>>::insert(index, current_block_number);
```

You need to declare the type of the keyValue to Rust. Also, SomethingConfigIndex is currently a 32 bit.